### DISCUSSION TITLE
Can the depreciation of fwknop be reconsidered?

### DISCUSSION THREAD

**Discussion by theory16** [Homebrew/discussions#5554] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.14-40-gd40e170
ORIGIN: https://github.com/Homebrew/brew
HEAD: d40e1707f2cf1847fc8481796b6706ea18065482
Last commit: 25 hours ago
Core tap HEAD: 65c7caba4864f09eba7f7583c46356c6ab9a8058
Core tap last commit: 24 hours ago
Core tap JSON: 11 Aug 17:43 UTC
Core cask tap HEAD: 672769ef7a4aee91ed12248345785998d4169ace
Core cask tap last commit: 24 hours ago
Core cask tap JSON: 11 Aug 17:43 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.nYoakL4fIF/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 12.7.6-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  pdftk

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  fwknop
  ilmbase
  mcrypt
  openssl@1.1

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-versions
Untap them with `brew untap`.

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  openssl@3
  keyring
  fwknop

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   M Formula/f/fwknop.rb
  ?? Formula/f/fwknop-ajs.rb
  ?? Formula/htdig.rb
```

### Description of issue

The formula fwknop is marked disabled because :unmaintained - which I read means the project is abandoned. However, the project source is being maintained (last update is 7th Feb) and version 2.6.11. The client side of it is much used by me. I can compile it from source and use it outside home-brew but I suspect there are others who would value this back within homebrew. Any prospect of revisiting this decision? Thanks

**Comment by Bo98** [comment, marked as answer]
It was deprecated for one year and has been disabled for 10 months. The number of installs is very low and it seems that you're the first person in two years to notice.

The deprecation cycle started in 2022 in response to this: https://github.com/mrash/fwknop/issues/344#issuecomment-1264405600. A few months _after_ we started the deprecation cycle, a comment restarting development was made so we could reconsider deprecation should someone open a pull request to update it to 2.6.11.

**Reply by theory16** [reply]
Thanks. I obviously noticed as soon as it was marked deprecated, but thought the compiled version might just keep working. An update to another package last week (gpgme?) then seemed to remove/overwrite some libraries which meant my version stopped working - that's the point where my "noticing" become more active! Anyway, you may be right that I could be the only person who uses it (and I only need the client side for Macs) so perhaps not much incentive for this to be prioritised. I do appreciate the quick response. Thanks.

**Comment by ZhongRuoyu** [comment]
I have opened a PR to undeprecate it: https://github.com/Homebrew/homebrew-core/pull/181015

**Reply by ZhongRuoyu** [reply]
`fwknop` is now back to life.

**Reply by theory16** [reply]
Thank so much
